test/sharness/lib/test-lib.sh

```diff
@@ -45,7 +45,9 @@ test "$TEST_EXPENSIVE" = 1 && test_set_prereq EXPENSIVE
 test "$TEST_NO_DOCKER" != 1 && type docker && test_set_prereq DOCKER
 
 TEST_OS=$(uname -s | tr [a-z] [A-Z])
-test expr "$TEST_OS" : "CYGWIN_NT" >/dev/null && test_set_prereq POSIX
+
+# Set a prereq as error messages are often different on Windows/Cygwin
+expr "$TEST_OS" : "CYGWIN_NT" >/dev/null || test_set_prereq STD_ERR_MSG
```

Comment:
So what about a name like "LINUX_LIKE_ERRORS"?

I think this is both more descriptive than "STD_ERR_MSG" and more future proof than "LINUX_OR_OSX". 

Also I like that it is specific to errors because when differences are not specific to errors we should not use a prereq but instead do something like what we do in `do_umount()` in "test-lib.sh":

```
do_umount() {
    if [ "$(uname -s)" = "Linux" ]; then
	fusermount -u "$1"
    else
	umount "$1"
    fi
}
```